### PR TITLE
Add option to pretty print json files in compiler

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

For debug purpose, kani-compiler can now generate json files that are more human friendly.

By default this is disabled, since computers don't care about pretty printing and it takes a lot of space.

### Resolved issues:

N/A

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/kani-compiler/src/parser.rs [hunk 1]
```diff
@@ -127,6 +130,11 @@ pub fn parser<'a, 'b>() -> App<'a, 'b> {
                 .long("--assertion-reach-checks")
                 .help("Check the reachability of every assertion."),
         )
+        .arg(
+            Arg::with_name(PRETTY_OUTPUT_FILES)
+                .long("--pretty-files")
+                .help("Output json files will be more human friendly (with spaces)."),
```
Comment (zhassan-aws) [inline src/kani-compiler/src/parser.rs:136]:
```suggestion
                .help("Output json files in a more human-readable format (with spaces)."),
```

File: src/kani-compiler/src/parser.rs [hunk 2]
```diff
@@ -127,6 +130,11 @@ pub fn parser<'a, 'b>() -> App<'a, 'b> {
                 .long("--assertion-reach-checks")
                 .help("Check the reachability of every assertion."),
         )
+        .arg(
+            Arg::with_name(PRETTY_OUTPUT_FILES)
+                .long("--pretty-files")
```
Comment (zhassan-aws) [inline src/kani-compiler/src/parser.rs:135]: I suggest adding "json" somewhere in the option (e.g. pretty-json-files) to clarify that this is specific for json files.

**Review by zhassan-aws — COMMENTED** [review]
How is this option intended to be used (I don't see it hooked up to `cargo-kani`)?

**Comment by celinval** [comment]
> 

It's not. This is for debug purpose only. One would need to run the compiler bit manually.

File: src/kani-compiler/src/parser.rs (hunk 2 above)
Comment (celinval) [reply src/kani-compiler/src/parser.rs:135]: Ok

**Comment by celinval** [comment]
BTW, I can add a hidden switch to cargo-kani though.

**Comment by zhassan-aws** [comment]
> BTW, I can add a hidden switch to cargo-kani though.

I'm fine either way (now that you've explained how you intend to use it). For the record, I've been using `--keep-temps` along with `python3 -m json.tool <json-file> > <pretty-printed-json-file>` on my end for debugging purposes. Not sure how this compares to the output from `serde_json::to_writer_pretty` though.

**Comment by celinval** [comment]
> > BTW, I can add a hidden switch to cargo-kani though.
> 
> I'm fine either way (now that you've explained how you intend to use it). For the record, I've been using `--keep-temps` along with `python3 -m json.tool <json-file> > <pretty-printed-json-file>` on my end for debugging purposes. Not sure how this compares to the output from `serde_json::to_writer_pretty` though.

I usually do that too but sometimes it fails for me. I don't know why. Maybe I have to update my json module.

**Review by zhassan-aws — APPROVED** [review]
Thanks!

**Pull request merged**